src/lib.rs

```diff
@@ -25,50 +29,84 @@ impl PyEvtxParser {
 
         obj.init({
             PyEvtxParser {
-                inner: inner.into_chunks(),
-                records: None,
-                settings: ParserSettings::new(),
+                inner: Some(inner)
             }
         });
 
         Ok(())
     }
+
+    fn records(&mut self) -> PyResult<PyRecordsIterator> {
+        self.records_iterator(OutputFormat::XML)
+    }
+
+    fn records_json(&mut self) -> PyResult<PyRecordsIterator> {
+        self.records_iterator(OutputFormat::JSON)
+    }
 }
 
 impl PyEvtxParser {
-    fn err_to_object(e: failure::Error, py: Python) -> PyObject {
-        PyErr::new::<RuntimeError, _>(format!("{}", e)).to_object(py)
+    fn records_iterator(&mut self, output_format: OutputFormat) -> PyResult<PyRecordsIterator> {
+        let inner = match self.inner.take() {
+            Some(inner) => inner,
+            None => {
+                return Err(PyErr::new::<RuntimeError, _>("PyEvtxParser can only be used once"));
+            }
+        };
+
+        Ok(PyRecordsIterator {
+            inner: inner.into_chunks(),
+            records: None,
+            settings: ParserSettings::new(),
+            output_format: output_format,
+        })
     }
+}
 
-    fn record_to_pydict(gil: Python, record: SerializedEvtxRecord) -> PyResult<&PyDict> {
-        let pyrecord = PyDict::new(gil);
 
-        pyrecord.set_item("event_record_id", record.event_record_id)?;
-        pyrecord.set_item("timestamp", format!("{}", record.timestamp))?;
-        pyrecord.set_item("data", record.data)?;
-        Ok(pyrecord)
-    }
+fn err_to_object(e: failure::Error, py: Python) -> PyObject {
+    PyErr::new::<RuntimeError, _>(format!("{}", e)).to_object(py)
+}
 
-    fn record_to_pyobject(r: Result<SerializedEvtxRecord, failure::Error>) -> PyObject {
-        let gil = Python::acquire_gil();
-        let py = gil.python();
+fn record_to_pydict(gil: Python, record: SerializedEvtxRecord) -> PyResult<&PyDict> {
+    let pyrecord = PyDict::new(gil);
 
-        match r {
-            Ok(r) => match Self::record_to_pydict(py, r) {
-                Ok(dict) => dict.to_object(py),
-                Err(e) => e.to_object(py),
-            },
-            Err(e) => PyEvtxParser::err_to_object(e, py),
-        }
+    pyrecord.set_item("event_record_id", record.event_record_id)?;
+    pyrecord.set_item("timestamp", format!("{}", record.timestamp))?;
+    pyrecord.set_item("data", record.data)?;
+    Ok(pyrecord)
+}
+
+fn record_to_pyobject(r: Result<SerializedEvtxRecord, failure::Error>) -> PyObject {
```

Comment:
Use `evtx` version 0.3.0, errors are no longer using failure, so you can remove that and use `evtx::err::Error` (which doesn't require any dependency).